language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/util/documentsymbol/DocumentSymbolResolver.java

```diff
@@ -151,20 +177,146 @@
+        Optional<MetadataNode> metadata = moduleVariableDeclarationNode.metadata();
+        boolean isDeprecated = metadata.isPresent() &&
+                DocumentSymbolUtil.isDeprecated(metadata.get());
+        return Optional.ofNullable(createDocumentSymbol(name, symbolKind, null,
+                range, range, isDeprecated, Collections.emptyList(), this.context));
+    }
+
+    @Override
+    public Optional<DocumentSymbol> transform(ConstantDeclarationNode constantDeclarationNode) {
+        String name = constantDeclarationNode.variableName().text();
+        SymbolKind symbolKind = SymbolKind.Constant;
+        Range range = DocumentSymbolUtil.generateNodeRange(constantDeclarationNode);
+        Optional<MetadataNode> metadata = constantDeclarationNode.metadata();
+        boolean isDeprecated = metadata.isPresent() &&
+                DocumentSymbolUtil.isDeprecated(metadata.get());
+        return Optional.ofNullable(createDocumentSymbol(name, symbolKind, null,
+                range, range, isDeprecated, Collections.emptyList(), this.context));
+    }
+
+    @Override
+    public Optional<DocumentSymbol> transform(EnumDeclarationNode enumDeclarationNode) {
+        String name = enumDeclarationNode.identifier().text();
+        SymbolKind symbolKind = SymbolKind.Enum;
+        Range range = DocumentSymbolUtil.generateNodeRange(enumDeclarationNode);
+        Optional<MetadataNode> metadata = enumDeclarationNode.metadata();
+        boolean isDeprecated = metadata.isPresent() &&
+                DocumentSymbolUtil.isDeprecated(metadata.get());
+        return Optional.ofNullable(createDocumentSymbol(name, symbolKind, null,
+                range, range, isDeprecated, Collections.emptyList(), this.context));
+    }
+
+    @Override
+    public Optional<DocumentSymbol> transform(ModuleXMLNamespaceDeclarationNode moduleXMLNamespaceDeclarationNode) {
+        String name = SyntaxKind.XMLNS_KEYWORD.stringValue() + " "
+                + moduleXMLNamespaceDeclarationNode.namespaceuri().toSourceCode();
+        Optional<IdentifierToken> prefix = moduleXMLNamespaceDeclarationNode.namespacePrefix();
+        if (prefix.isPresent()) {
+            name = name + " as " + prefix.get();
+        }
+        SymbolKind symbolKind = SymbolKind.Namespace;
+        Range range = DocumentSymbolUtil.generateNodeRange(moduleXMLNamespaceDeclarationNode);
+        return Optional.ofNullable(createDocumentSymbol(name, symbolKind, null,
+                range, range, false, Collections.emptyList(), this.context));
+    }
+
+    @Override
+    public Optional<DocumentSymbol> transform(ListenerDeclarationNode listenerDeclarationNode) {
+        StringBuilder name = new StringBuilder(SyntaxKind.LISTENER_KEYWORD.stringValue());
+        Optional<TypeDescriptorNode> typeDescriptorNode = listenerDeclarationNode.typeDescriptor();
+        if (typeDescriptorNode.isPresent()) {
+            name.append(" " + typeDescriptorNode.get().toSourceCode());
```

Comment:
Do we need to add the listener keyword and the type descriptor as the name?

Reply:
My intention was to add service keyword to service objects and listener keyword to listener objects. The type descriptors are not necessary as you suggested. Let's get rid of them. WDYT about the service declaration? should we keep the service keyword there?

Reply:
Since the services do not have a symbol name associated with it, it's okay to generate the name. For any of the symbols with a name or a prefix as in xml namespace declaration, we don't need to modify the identifier.